### PR TITLE
Rework face API and fix transports

### PR DESCRIPTION
**Pull request by pulsejet**
Replace #66

- [x] Improve link service API (unnecessary goroutines)
- [x] Get PIT token functional again
- [x] Remove complex HasQuit signalling
- [x] Internal transport
- [x] Unix transport
- [x] Unix listener
- [x] UDP transport
- [x] UDP listener
- [x] TCP transport
- [x] TCP listener
- [x] TCP persistency
- [x] Websocket transport
- [x] Websocket listener
- [x] Multicast UDP transport
- [ ] Ethernet transport (?)
- [x] On demand face expiry

### CODE REVIEW

**Comment by pulsejet** [comment]
Done with this, not doing ethernet at the moment

File: face/table.go [hunk 1]
```diff
@@ -80,3 +82,21 @@ func (t *Table) Remove(id uint64) {
...
 }
+
+// ExpirationHandler stops the faces that have expired
+func (t *Table) ExpirationHandler() {
+	for {
+		// Check for expired faces every 10 seconds
+		time.Sleep(10 * time.Second)
+
+		// Iterate the face table
+		t.faces.Range(func(_, face interface{}) bool {
```
Comment (zjkmxy) [inline face/table.go:93]: (Not related) Really want to see generics supported in `sync` package, but it seems to still have a long way to go: golang/go#47657

File: face/multicast-udp-transport.go [hunk 2]
```diff
@@ -65,32 +62,51 @@ func MakeMulticastUDPTransport(localURI *defn.URI) (*MulticastUDPTransport, erro
...
 
-	t.changeState(defn.Up)
-
 	return t, nil
 }
 
+func (t *MulticastUDPTransport) connectSend() error {
+	sendConn, err := t.dialer.Dial(t.remoteURI.Scheme(), t.groupAddr.String())
+	if err != nil {
+		return errors.New("Unable to create send connection to group address: " + err.Error())
```
Comment (zjkmxy) [inline face/multicast-udp-transport.go:87]: (`ST1005`) Go prefers lowercased error message.

**Review by zjkmxy — APPROVED** [review]
LGTM for now

**Comment by zjkmxy** [comment]
I may not have time to do a test run this weekends so I will not initiate merging. Varun, please merge it yourself after test and when you think it is ready.

File: face/table.go (hunk 1 above)
Comment (pulsejet) [reply face/table.go:93]: Yeah, this is very annoying

**Comment by pulsejet** [comment]
> I may not have time to do a test run this weekends so I will not initiate merging. Varun, please merge it yourself after test and when you think it is ready.

No worries, thanks again for the review :D

**Pull request merged**